aria-practices.html

```diff
@@ -6279,6 +6279,38 @@ <h3>Indicating sort order with <code>aria-sort</code></h3>
 
   </section>
 
+  <section id="accessibility_tree">
+    <h2>Accessibility tree</h2>
+    <p>Browsers use a so-called accessibility tree to communicate the representation of web content with assistive technologies. See the <a href="https://w3c.github.io/core-aam/#dfn-accessibility-tree">Core Accessibility API Mappings</a> specification for further information about accessibility trees.</p>
+    <p>The accessibility tree is based on information in the DOM tree and information in CSS. For example, text in the DOM tree can also be in the accessibility tree. CSS generated content is not part of the DOM tree, but can be part of the accessibility tree.</p>
```

Comment:
> The accessibility tree is based on information in the DOM tree and information in CSS.

Should we also say, " and ARIA"? As in:
> The accessibility tree is based on information in the DOM tree and information in CSS and ARIA.

Reply:
ARIA is in the DOM tree isn't it?

Reply:
Will it be later when authors can directly send it via JavaScript (AOM)?
I'm not clear on whether AOM-changes will be reflected in the DOM or not.

Reply:
@jnurthen 
> ARIA is in the DOM tree isn't it?

Good point.  :)

@StommePoes
> Will it be later when authors can directly send it via JavaScript (AOM)?
> I'm not clear on whether AOM-changes will be reflected in the DOM or not.

I'm not sure whether all of the upcoming AOM-changes will be reflected in the DOM (@alice?), however I recently tested some simple attribute reflection in pull https://github.com/w3c/aria-practices/pull/1161 and I can confirm that ARIA attributes set using dot notation are definitely reflected in the DOM. (yay!)

Reply:
Yay indeed!

For **[element reflection](https://github.com/WICG/aom/blob/gh-pages/explainer.md#reflecting-element-references)**, we will set the content attribute to the empty string if it's not possible to set it to anything meaningful (e.g. if you pass in an element with no ID, etc.) 

So you will have a "hint" in the DOM that you need to check via the property:

Say you had:
```html
<input id="input"></input>
<optgroup id="group">
  <option>Option 1</option>
  <option>Option 2</option>
</optgroup>
```

Then you did:

```js
input.ariaActiveDescendantElement = group.firstElementChild;
```

Then you would have

```html
<input id="input" aria-activedescendant></input>
<!-- ... -->
```

- but retrieving `input.ariaActiveDescendantElement` would return the first `<option>`.

For [**Custom Element default semantics**](https://github.com/WICG/aom/blob/gh-pages/explainer.md#default-semantics-via-customelementsdefine), it's a little trickier - just like for native elements, there won't be a straightforward way to retrieve the "internal" properties.

If a Custom Element's implementation does this:

```js
this._internals = customElements.createInternals(this);
this._internals.role = "tab";
this._internals.ariaSelected = false;
this._internals.ariaControlsElement = tabPanel;
```

then a script in the page context has no reliable way to know about that, since it is opaque to the page (again, like native elements).

For **[Virtual Accessibility Nodes](https://github.com/WICG/aom/blob/gh-pages/explainer.md#virtual-accessibility-nodes)**, absolutely nothing has been decided for sure, but it seems likely that they will not be DOM nodes and will only be available to script.

Reply:
Oof. Off-topic I know, but recently auditing and checking that aria-current was being (re) set correctly as user moved thru a nav on an SPA. Could only see that in the DOM, did not show up in any browser's tree, and SRs were contradicting each other :( Hope extra tools come with AOM changes.

Reply:
i.e. extra testing/debugging tools to programmatically probe the accessibility tree?

Reply:
(off-topic reply) FF seems to expose current in the a11y tree correctly as far as I can see. It would be nice if chrome exposed all ARIA attributes in the "ARIA Attributes" section of the Accessibility pane (I suspect it is simply an oversight that aria-current is not there and a bug needs to be logged)